### PR TITLE
Make qr code arguments overrideable in TwoFactorSetup.

### PR DESCRIPTION
**Pull request by valberg**
Addresses #53

### CODE REVIEW

**Comment by akx** [comment]
Heh, GMTA – I just wrote #157 for the same thing...

**Comment by valberg** [comment]
> Heh, GMTA – I just wrote #157 for the same thing...

Hah hah very nice! You pick which version to go for - I will not be offended if you pick yours :)

File: allauth_2fa/views.py [hunk 1]
```diff
@@ -84,8 +85,28 @@ def get(self, request, *args, **kwargs):
...
 
+    def get_qr_code_kwargs(self):
+        """Keyword arguments to give to the QR code function.
+
+        This is a separate method so that it can be overridden if needed.
+
+        The fields are:
+        - issuer: The name of the site (e.g. "example.com")
+        - label: The label for the QR code (e.g. "example.com: username")
+        """
```
Comment (akx) [inline allauth_2fa/views.py]: The docstring could use some tweaking, and we can just as well be helpful and paraphrase [the spec for `otpauth` URLs](https://github.com/google/google-authenticator/wiki/Key-Uri-Format).

Let's slap a type annotation on while we're at it.

```suggestion
    def get_qr_code_kwargs(self) -> dict[str, str]:
        """
        Get the configuration for generating a QR code.

        The fields required are:
        - `label`: identifies which account a key is associated with. Contains an account name, preferably prefixed by an issuer name and a colon, e.g. `issuer: account`.
        - `issuer`: indicates the provider or service this account is associated with.
        """
```

**Review by akx — CHANGES_REQUESTED** [review]

File: allauth_2fa/views.py [hunk 2]
```diff
@@ -84,8 +85,29 @@ def get(self, request, *args, **kwargs):
         self._new_device()
         return super().get(request, *args, **kwargs)
 
+    def get_qr_code_kwargs(self) -> dict[str, str]:
```
Comment (akx) [inline allauth_2fa/views.py:90]: Ah, we'll need a `from __future__ import annotations` in this file for 3.7 :)

**Comment by valberg** [comment]
@akx Totally agree with adding typing - but we're going to have to do something about supporting Python 3.8 and below since https://peps.python.org/pep-0585/ is first implemented in 3.9

**Comment by akx** [comment]
> @akx Totally agree with adding typing - but we're going to have to do something about supporting Python 3.8 and below since https://peps.python.org/pep-0585/ is first implemented in 3.9

`from __future__ import annotations` ✨

**Review by akx — APPROVED** [review]

**Pull request merged**